docs/guides/code/request_storage_rl_with_crawler.py

```diff
@@ -0,0 +1,35 @@
+import asyncio
+
+from crawlee.http_crawler import HttpCrawler, HttpCrawlingContext
+from crawlee.storages import RequestList
+
+
+async def main() -> None:
+    # Open the request list, if it does not exist, it will be created.
+    # Leave name empty to use the default request list.
+    request_list = RequestList(
+        name='my-request-list',
+        requests=['https://apify.com/', 'https://crawlee.dev/'],
+    )
+
+    # Create a new crawler (it can be any subclass of BasicCrawler) and pass the request
+    # list as request provider to it. It will be managed by the crawler.
+    crawler = HttpCrawler(request_provider=request_list)
+
+    # Define the default request handler, which will be called for every request.
+    @crawler.router.default_handler
+    async def request_handler(context: HttpCrawlingContext) -> None:
+        context.log.info(f'Processing {context.request.url} ...')
+
+        # Use context's add_requests method helper to add new requests from the handler.
```

Comment:
I know there is a separate issue for that (https://github.com/apify/crawlee-python/issues/355), but shouldn't we explain why we use `add_requests` here? And maybe mention that `enqueue_links` is an alternative?

Reply:
I added a new section explaining the difference between these two.